### PR TITLE
Delete job role

### PR DESCRIPTION
**Pull request by AdamB-Kainos**

### CODE REVIEW

File: views/pages/allJobRolesList.html
Comment (jackpollinsK) [inline views/pages/allJobRolesList.html:1]: add in id to assist in testing

File: views/pages/allJobRolesList.html [hunk 1]
```diff
@@ -1,36 +1,42 @@
...
-            <td>{{ jobRole.band }}</td>
-            <td>{{ jobRole.closingDate | date }}</td>
-            {% if userLevel.Role == 2 %}
-                <td><a href="/job-apply-{{jobRole.id}}" class="btnKainos">Apply Now</a></td>
-            {% endif %}
-        </tr>
-    {% endfor %}
-    </tbody>
-</table>
+    <div class="container m-5">
```
Comment (jackpollinsK) [inline views/pages/allJobRolesList.html:4]: add in id to help with testing

**Review by jackpollinsK — CHANGES_REQUESTED** [review]

**Review by jackpollinsK — APPROVED** [review]

File: src/main/controllers/JobRoleController.ts [hunk 2]
```diff
@@ -8,15 +8,26 @@ export const getAllJobRoles = async (req: express.Request, res: express.Response
...
+        res.locals.errormessage = e.message;
+        res.locals.pageName = "An Error Ocured";
+        res.render("pages/errorPage.html", res);
+    }
+}
+
+export const deleteJobRole = async (req: express.Request, res: express.Response): Promise<void> => {
+    try{
+        await deleteJobRoleById(req.params.id,req.session.token)
+        res.render('pages/allJobRolesList.html', {jobRoles: await getJobRoles(req.session.token), pageName: "Job Roles", token: req.session.token, userLevel: jwtDecode(req.session.token)});
```
Comment (shaunganley) [inline src/main/controllers/JobRoleController.ts]: you could redirect here instead

File: src/main/controllers/JobRoleController.ts [hunk 3]
```diff
@@ -8,15 +8,26 @@ export const getAllJobRoles = async (req: express.Request, res: express.Response
...
     }catch (e) {
         res.locals.errormessage = e.message;
         res.locals.pageName = "An Error Ocured";
```
Comment (shaunganley) [inline src/main/controllers/JobRoleController.ts]: pretty sure occurred is misspelt across this file

File: src/test/unit/controller/JobRoleControllerTests.ts [hunk 4]
```diff
@@ -187,4 +181,24 @@ describe('JobRoleController', function () {
...
+            };
+
+            sinon.stub(JobRoleService, 'getJobRoles');
+            sinon.stub(JobRoleService, 'deleteJobRoleById');
+
+            await JobRoleController.deleteJobRole(req as unknown as express.Request, res as unknown as express.Response);
+
+            expect(res.render.called).to.be.true;
+            expect(res.render.calledWith('pages/allJobRolesList.html')).to.be.true;
+        });
```
Comment (shaunganley) [inline src/test/unit/controller/JobRoleControllerTests.ts:222]: need another test for error being throw

**Review by shaunganley — CHANGES_REQUESTED** [review]

**Review by shaunganley — APPROVED** [review]

**Pull request merged**